dojo/importers/default_reimporter.py

```diff
@@ -286,13 +351,21 @@ def close_old_findings(
                     finding,
                     f"Mitigated by {self.test.test_type} re-upload.",
                     finding_groups_enabled=self.findings_groups_enabled,
+                    product_grading_option=False,
                 )
                 mitigated_findings.append(finding)
         # push finding groups to jira since we only only want to push whole groups
         if self.findings_groups_enabled and self.push_to_jira:
             for finding_group in {finding.finding_group for finding in findings if finding.finding_group is not None}:
                 jira_helper.push_to_jira(finding_group)
 
+        # Calculate grade once after all findings have been closed
+        if mitigated_findings:
+            product = self.test.engagement.product
+            system_settings = System_Settings.objects.get()
+            if system_settings.enable_product_grade:
+                calculate_grade(product)
```

Comment:
This block of code is written four times total. Can it moved to the base importer?